Automattic/jetpack
# Search: Add scaffolding for Gutenberg-enabled Customizer

Split off from #20209.

#### Changes proposed in this Pull Request:
* Adds a blank page React application at `/wp-admin/admin.php?page=jetpack-search-customize`.
  * This will be the new home of a new React application that provides a Gutenberg-styled search customization experience, to be added in a follow-up PR.
  * Since this page is only accessible via manual URL manipulation, it should be safe to ship as-is.
* Updates `@wordpress/dependency-extraction-webpack-plugin` package to ensure compatibility with `@wordpress/interface`, which Customberg will rely on.
* Updates the search app build process, adds a new `watch-search` command.

#### Jetpack product discussion
See pcNPJE-7J-p2, #20128, and #20209.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
Apply these changes to your site with a Jetpack Search subscription.

* Testing build and dependency updates:
  * Ensure that all existing JS tests pass as expected.
  * Ensure that `pnpm build` and `pnpm watch` work as expected.
  * Ensure that `pnpm build-search` and `pnpm watch-search` work as expected.

* Testing the new placeholder page for "Customberg":
  * Navigate to the new customization page at `/wp-admin/admin.php?page=jetpack-search-customize`.
  * Ensure that the page renders a blank (albeit broken-looking) page that says "Customberg is here!".
